### PR TITLE
To Do List: CRUD

### PR DESCRIPTION
**Pull request by TadesseDev**
# Changes introduced
- Remove all hardcoded items from the task array. ✔
- Create new JavaScript files (modules) for the new functionality. ✍🏼
- Implement a functions for CRUD (Create, Read, Update, Delete). 😎
- Rearange task index on add and remove (insert to the end and update index of all tasks on delete). ✔
- All changes to the To Do List saved in local storage. 🎞

### CODE REVIEW

File: src/modules/MyTasks.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      task.index = i;
+      i += 1;
+    });
+    alterBookList(MyTasks.tasks);
+  }
+
+  static addItem = (task) => {
+    MyTasks.tasks.push(task);
+    alterBookList(MyTasks.tasks);
+  }
```
Comment (Gambit142) [inline src/modules/MyTasks.js:38]: - [ ] Nice work you have done here. This is great. When adding a task, the index of the task should be set to the length of the tasks array. That means the first task should have its index equal to `1` the second task equal to `2` and so on. Kindly implement it. `task.index = MyTasks.tasks.length + 1;`

File: src/modules/MyTasks.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  static removeItem = (removeTask) => {
+    const temp = [];
+    for (let i = 0; i < MyTasks.tasks.length; i += 1) {
+      if (removeTask !== MyTasks.tasks[i]) {
+        temp.push(MyTasks.tasks[i]);
+      }
+    }
+    MyTasks.tasks = temp;
+    // update task index on each remove
+    let i = 0;
```
Comment (Gambit142) [inline src/modules/MyTasks.js]: - [ ] Kindly set `i` to 1. Remember that the index of the task should equal the length of the tasks array. Kindly implement it. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Tadesse-Alemayehu 👋, 

Good job so far!

### Highlights 
- Linters checks passed ✔️ 
- Good Use of modules to separate concerns ✔️ 
- Clean and dry code ✔️ 

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- In your _**Getting Started**_ section of your `README.md` file, you can consider including commands like `npm install` to install all dependencies of the project after cloning it on the local machine and `npm start` to start the server after installing the various dependencies. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/modules/MyTasks.js (hunk 2 above)
Comment (TadesseDev) [reply src/modules/MyTasks.js]: @Gambit142 Nice catch 😀. thanks for the thorough review you provided. I’ll make the changes before requesting a new code review. ✋🏼

**Review by rustam-tolipov — APPROVED** [review]
## Hello @Tadesse-Alemayehu 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉, 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by TadesseDev** [comment]
@Rustamxon7 Thank you.

File: src/modules/MyTasks.js (hunk 2 above)
Comment (TadesseDev) [reply src/modules/MyTasks.js]: Dear @Gambit142, I have implemented the required changes in [9b0c028](https://github.com/Tadesse-Alemayehu/To-Do-list/pull/6/commits/9b0c02853d82fa375b7f1e0eb7357c1c370532f2) commit hash and optional changes in [a37dc07](https://github.com/Tadesse-Alemayehu/To-Do-list/pull/6/commits/a37dc073f30b6393a62690c32b7a504092888d65) commit hash. Thank you!

Comment (Gambit142) [reply src/modules/MyTasks.js]: Hi @Tadesse-Alemayehu, nice work and professionalism. Keep up the hard work. 👍

**Pull request merged**